lib/queue.js

```diff
@@ -642,32 +642,19 @@ Queue.prototype.startMoveUnlockedJobsToWait = function() {
   }
 };
 
-
-Queue.prototype.processJobs = function(resolve, reject, options){
+Queue.prototype.processJobs = function(resolve, reject){
   var _this = this;
-  var processJobs = this.processJobs.bind(this, resolve, reject, options);
+  var processJobs = this.processJobs.bind(this, resolve, reject);
   if(!this.closing){
-    var idleTime = 0;
-    var obeyRateLimit = function() {};
-
-    if (options.rateLimiter) {
-      obeyRateLimit = function() {
-        return options.rateLimiter()
-          .then(function(timeLeft) {
-            if (!timeLeft) { return; }
-            idleTime += timeLeft;
-            return Promise.delay(timeLeft)
-              .then(obeyRateLimit);
-          });
-      };
-    }
+    var startTime = new Date().getTime();
 
     process.nextTick(function(){
       (_this.paused || Promise.resolve())
-        .then(obeyRateLimit)
         .then(_this.getNextJob)
         .then(function(job) {
-          job.idleTime = idleTime;
+          if (job !== undefined) {
```

Comment:
another thing I am wondering, what does idleTime represent here? Because when startTime is defined, there may not even jobs in the queue, so it could happen that long time passes before a job arrives and gets processed, which will set idleTime to a large value although the job was processed immediately.